packages/aws-cdk-lib/core/test/removal-policies.test.ts

```diff
@@ -0,0 +1,252 @@
+  public static readonly CFN_RESOURCE_TYPE_NAME = 'AWS::Test::Resource';
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      type: TestResource.CFN_RESOURCE_TYPE_NAME,
+    });
+  }
+}
+
+class TestBucketResource extends CfnResource {
+  public static readonly CFN_RESOURCE_TYPE_NAME = 'AWS::S3::Bucket';
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      type: TestBucketResource.CFN_RESOURCE_TYPE_NAME,
+    });
+  }
+}
+
+class TestTableResource extends CfnResource {
+  public static readonly CFN_RESOURCE_TYPE_NAME = 'AWS::DynamoDB::Table';
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      type: TestTableResource.CFN_RESOURCE_TYPE_NAME,
+    });
+  }
+}
+
+describe('removal-policys', () => {
+  test('applies removal policy to all resources in scope', () => {
+    // GIVEN
+    const stack = new Stack();
+    const parent = new Construct(stack, 'Parent');
+    const resource1 = new TestResource(parent, 'Resource1');
+    const resource2 = new TestResource(parent, 'Resource2');
+
+    // WHEN
+    RemovalPolicies.of(parent).destroy();
+
+    // THEN
+    synthesize(stack);
+    expect(resource1.cfnOptions.deletionPolicy).toBe('Delete');
+    expect(resource2.cfnOptions.deletionPolicy).toBe('Delete');
+  });
+
+  test('applies removal policy only to specified resource types using strings', () => {
+    // GIVEN
+    const stack = new Stack();
+    const parent = new Construct(stack, 'Parent');
+    const bucket = new TestBucketResource(parent, 'Bucket');
+    const table = new TestTableResource(parent, 'Table');
+    const resource = new TestResource(parent, 'Resource');
+
+    // WHEN
+    RemovalPolicies.of(parent).retain({
+      applyToResourceTypes: [
+        bucket.cfnResourceType, // 'AWS::S3::Bucket'
+        TestTableResource.CFN_RESOURCE_TYPE_NAME, // 'AWS::DynamoDB::Table'
+      ],
+    });
+
+    // THEN
+    synthesize(stack);
+    expect(bucket.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(table.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(resource.cfnOptions.deletionPolicy).toBeUndefined();
+  });
+
+  test('applies removal policy only to specified resource types using classes', () => {
+    // GIVEN
+    const stack = new Stack();
+    const parent = new Construct(stack, 'Parent');
+    const bucket = new TestBucketResource(parent, 'Bucket');
+    const table = new TestTableResource(parent, 'Table');
+    const resource = new TestResource(parent, 'Resource');
+
+    // WHEN
+    RemovalPolicies.of(parent).retain({
+      applyToResourceTypes: [
+        TestBucketResource.CFN_RESOURCE_TYPE_NAME, // 'AWS::S3::Bucket'
+        table.cfnResourceType, // 'AWS::DynamoDB::Table'
+      ],
+    });
+
+    // THEN
+    synthesize(stack);
+    expect(bucket.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(table.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(resource.cfnOptions.deletionPolicy).toBeUndefined();
+  });
```

Comment:
Since we have dropped specifying the Class type for `applyToResourceTypes` and unified it with string type, it is good to remove this test.

```suggestion
```